// astro/dist/assets/fonts/constants.js
const LOCAL_PROVIDER_NAME = "local";
const DEFAULTS = {
  weights: ["400"],
  styles: ["normal", "italic"],
  subsets: ["cyrillic-ext", "cyrillic", "greek-ext", "greek", "vietnamese", "latin-ext", "latin"],
  // Technically serif is the browser default but most websites these days use sans-serif
  fallbacks: ["sans-serif"],
  optimizedFallbacks: true
};
const VIRTUAL_MODULE_ID = "virtual:astro:assets/fonts/internal";
const RESOLVED_VIRTUAL_MODULE_ID = "\0" + VIRTUAL_MODULE_ID;
const URL_PREFIX = "/_astro/fonts/";
const CACHE_DIR = "./fonts/";
const FONT_TYPES = ["woff2", "woff", "otf", "ttf", "eot"];
const DEFAULT_FALLBACKS = {
  serif: ["Times New Roman"],
  "sans-serif": ["Arial"],
  monospace: ["Courier New"],
  cursive: [],
  fantasy: [],
  "system-ui": ["BlinkMacSystemFont", "Segoe UI", "Roboto", "Helvetica Neue", "Arial"],
  "ui-serif": ["Times New Roman"],
  "ui-sans-serif": ["Arial"],
  "ui-monospace": ["Courier New"],
  "ui-rounded": [],
  emoji: [],
  math: [],
  fangsong: []
};
const FONTS_TYPES_FILE = "fonts.d.ts";
export {
  CACHE_DIR,
  DEFAULTS,
  DEFAULT_FALLBACKS,
  FONTS_TYPES_FILE,
  FONT_TYPES,
  LOCAL_PROVIDER_NAME,
  RESOLVED_VIRTUAL_MODULE_ID,
  URL_PREFIX,
  VIRTUAL_MODULE_ID
};
